Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml

```diff
@@ -14,33 +14,34 @@ relevantTechniques:
   - T1571
   - T1496
 query: |
-
+  let lookback = 7d;
   AzureActivity
+  | where TimeGenerated >= ago(lookback)
   | where OperationNameValue has_any ("ipfilterrules", "securityRules", "publicIPAddresses", "firewallrules") and OperationNameValue endswith "write"
   // Choosing Accepted here because it has the Rule Attributes included
   | where ActivityStatusValue == "Accepted" 
   // If there is publicIP info, include it
-  | extend publicIPAddress_ = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).ipAddress) 
-  | extend publicIPAddressVersion_ = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).publicIPAddressVersion) 
-  | extend publicIPAllocationMethod_ = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).publicIPAllocationMethod) 
+  | extend publicIPAddressVersion = tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties.publicIPAddressVersion)
```

Comment:
Same value of responseBody is getting parsed multiple times. Can we do this once and use it in different locations?

Reply:
Thanks @devikamehra . This is done